crates/ra_lsp_server/src/world.rs

```diff
@@ -279,3 +282,35 @@ impl WorldSnapshot {
         self.analysis.feature_flags()
     }
 }
+
+#[cfg(target_os = "windows")]
+fn lowercase_drive_letter(url: &Url) -> Url {
+    use std::str::FromStr;
+
+    let s = url.to_string();
+    let drive_partition: Vec<&str> = s.rsplitn(2, ':').collect::<Vec<&str>>();
```

Comment:
I am slightly worried that we apply a blanket transform to all URIs. Should we check that the scheme is `file`? 

I think I would also prefer it this funciton weren't `cfg(windows)` -- part of the appeal of URIs is that they allow represent various weired paths, like a paths from something like wsl, or a path on the remote machine, so I think it might be the case that we see a windows path here even if the host machine is linux.

Reply:
I've pushed a change that will now do a quick scan to check if there are any `Drive` components in the path, and would otherwise fallback to the default `Url` constructor.

When using WSL, you can only access windows paths from `/mnt/c` (when going from wsl->windows), I've checked the wierder case, which is code stored on UNC shares.

which is something like `\\localhost\c$\my_dir`.

rust will not treat this as a drive (even though it is technically local), so it will simply pass through.

I'm not entierly sure what vscode will do when accessing code stored on shares like this, but I would guess it would simply pass it through similarly.